packages/cart-sdk/hooks/use-checkout.ts

```diff
@@ -0,0 +1,270 @@
+import { useShopifyStorefrontClient } from '@ifixit/shopify-storefront-client';
+import * as React from 'react';
+import { CartError } from '../utils';
+import { useCart } from './use-cart';
+
+const LOCAL_CHECKOUT_STORAGE_KEY = 'checkout';
+
+type UseCheckout = CheckoutState & {
+   redirectToCheckout: RedirectToCheckoutFn;
+   reset(): void;
+};
+
+interface CheckoutState {
+   error: CartError | null;
+   isRedirecting: boolean;
+}
+
+interface RedirectToCheckoutFn {
+   (): Promise<void>;
+}
+
+export function useCheckout(): UseCheckout {
+   const cart = useCart();
+   const user = useAuthenticatedUser();
+   const standardCheckout = useStandardCheckout();
+   const draftOrderCheckout = useDraftOrderCheckout();
+   const [state, dispatch] = React.useReducer(reducer, {
+      error: null,
+      isRedirecting: false,
+   });
+   return {
+      ...state,
+      redirectToCheckout: async () => {
+         if (cart.data == null || !cart.data.hasItemsInCart) {
+            dispatch({
+               type: ActionType.FailedCheckout,
+               error: CartError.EmptyCart,
+            });
+         } else {
+            dispatch({ type: ActionType.StartCheckout });
+            let url: string;
+            try {
+               if (user.data?.is_pro) {
+                  url = await draftOrderCheckout();
+               } else {
+                  url = await standardCheckout();
+               }
+               window.location.href = url;
+            } catch (error) {
+               if (isError(error)) {
+                  console.error(error.message);
+               }
+               dispatch({
+                  type: ActionType.FailedCheckout,
+                  error: CartError.UnknownError,
+               });
+            }
+         }
+      },
+      reset: () => {
+         dispatch({ type: ActionType.ResetCheckout });
+      },
+   };
+}
+
+enum ActionType {
+   StartCheckout,
+   FailedCheckout,
+   ResetCheckout,
+}
+
+type CheckoutAction =
+   | StartCheckoutAction
+   | FailedCheckoutAction
+   | ResetCheckoutAction;
+
+interface StartCheckoutAction {
+   type: ActionType.StartCheckout;
+}
+
+interface FailedCheckoutAction {
+   type: ActionType.FailedCheckout;
+   error: CartError;
+}
+
+interface ResetCheckoutAction {
+   type: ActionType.ResetCheckout;
+}
+
+function reducer(state: CheckoutState, action: CheckoutAction): CheckoutState {
+   switch (action.type) {
+      case ActionType.StartCheckout:
+         return { isRedirecting: true, error: null };
+      case ActionType.FailedCheckout:
+         return { isRedirecting: false, error: action.error };
+      case ActionType.ResetCheckout:
+         return { isRedirecting: false, error: null };
+      default:
+         return assertNever(action);
+   }
+}
+
+function useDraftOrderCheckout() {
+   const appContext = useAppContext();
+   const client = useIFixitApiClient();
+   const ssoRoute = `${appContext.ifixitOrigin}/User/sso/shopify/ifixit-test?checkout=1`;
```

Comment:
I knew I was missing a few things like these, this PR has been open for too long 😁 So this specific url in the ifixit.com implementation comes from the php code, and I'm not sure how it work exactly.
I guess I am supposed to just replace `ifixit-test` with whatever is the Shopify store we're talking to right?

Reply:
Also the primary domain for `ifixit-test` is `store.cominor.com`. Is there a way to construct the sso route using that? Otherwise I'd have to store `ifixit-test` along side `store.cominor.com` in Strapi

Reply:
> I guess I am supposed to just replace ifixit-test with whatever is the Shopify store we're talking to right?

I think so, yes.

> Is there a way to construct the sso route using that? Otherwise I'd have to store ifixit-test along side store.cominor.com in Strapi

Good question, I don't know the answer off the top of my head. @sterlinghirsh?

Reply:
@masonmcelvain what do you think about merging this PR and open an issue for this specific thing?